### PR TITLE
Fix `zk-explorer` deployment failing on Netlify

### PR DESCRIPTION
**Pull request by yurixander**
## Summary of changes

_Provide a detailed description of proposed changes._

- A `public` folder is required, otherwise the build fails.
- Added an empty public folder, using `.gitkeep` to be able to stage it into a commit.

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [ ] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [x] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by yurixander** [comment]
The Netlify deployment logs seems to be complaining of memory exhaustion now, instead of the missing `public` folder (which is the error I was getting locally). However, the other dapps seem to be build properly, so that suggests that it is an issue specifically regarding `zk-explorer`.

**Comment by yurixander** [comment]
I've increased the Node.js memory limit via `NODE_OPTIONS=--max_old_space_size=4096`, and the memory exhaustion error is gone. This is also present in the Netlify configuration of other dapps, such as `hubble-stats`.

The [deployment is now successful on Netlify](https://deploy-preview-1908--zkexplorer.netlify.app/).

**Comment by drewstone** [comment]
Looks good, there's a bit of a flash from light/dark theme immediately onload. That should get fixed up. Does this happen to you @yurixander ?

**Comment by yurixander** [comment]
> Looks good, there's a bit of a flash from light/dark theme immediately onload. That should get fixed up. Does this happen to you @yurixander ?

Yes it does happen to me! When I first started working on the application, there was a warning from Next.js about it, so I thought it was a side effect of Next.js non-production builds or something along those lines. I'll look into it more in detail to find what's the problem.

File: apps/zk-explorer/components/GitHubOAuthButton/GitHubOAuthButton.tsx [hunk 1]
```diff
@@ -6,36 +6,45 @@ import { twMerge } from 'tailwind-merge';
...
+  scope,
+  state,
+  doInterceptOauthRedirect,
+  username,
+  onOAuthSuccess,
+  onOAuthError,
+  onSignedInClick,
+  ...rest
+}) => {
   const handleClick: MouseEventHandler<HTMLButtonElement> = (e) => {
```
Comment (vutuanlinh2k2) [inline apps/zk-explorer/components/GitHubOAuthButton/GitHubOAuthButton.tsx:21]: update this to UseCallback function

**Review by vutuanlinh2k2 — CHANGES_REQUESTED** [review]
After changing this you can merge to the `develop` branch right away @yurixander

**Pull request merged**